### ISSUE TITLE
WSL2-Error: A connection attempt failed on Windows 19555.1001

### ISSUE THREAD

**Issue by ad-on-is** [microsoft/WSL#4860]
After updating to Windows 19555 I get the following error, when starting WSL:

`A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`

```
PS C:\Users\add> wsl -l
Windows Subsystem for Linux Distributions:
Ubuntu-19.10 (Default)
docker-desktop
docker-desktop-data
PS C:\Users\add> wsl -d Ubuntu-19.10
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
PS C:\Users\add> wsl -d docker-desktop
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
PS C:\Users\add>
```

## Update ##
I wanted to try this workaround on a fresh Alpine, https://github.com/microsoft/WSL/issues/4371#issuecomment-521107556 but neither `/bin` nor `/sbin` were symlinks.

**Comment by olafkfreund** [comment]
Same problem here. All the WSL1 instances are working.

**Comment by gel-mashup** [comment]
i got this bug, i am using ubuntu 18.04 distro and i cant seem to start wsl2 instance

**Comment by FabienD** [comment]
Hi,

+1, VSCode can't established connection.

- VSCode Version: 1.42.0-insiders or stable 1.41.1
- Local OS Version: Microsoft Windows [Version 10.0.19555.1001]
- Remote OS Version: Ubuntu 18.04
- Remote Extension/Connection Type: WSL
- WSL Version: 2

**Comment by FabienD** [comment]
Hi,

For me, uninstalled / re-install **openssh-server** on the linux host fix the connection issue.

cf. https://docs.microsoft.com/fr-fr/windows/wsl/troubleshooting

**Comment by benhillis** [comment]
Thanks, we are debugging this internally - trying to figure out why only some users are hitting this.

**Comment by bjd145** [comment]
i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved. 

PS C:\Users\brian> wsl --list -v
  NAME            STATE           VERSION
* Ubuntu-18.04    Stopped         2
PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
PS C:\Users\brian> wsl --list -v
  NAME            STATE           VERSION
* Ubuntu-18.04    Running         1
PS C:\Users\brian>wsl
brian@brian_pc:~$

**Comment by ad-on-is** [comment]
> i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved.
> 
> PS C:\Users\brian> wsl --list -v
> NAME STATE VERSION
> 
> * Ubuntu-18.04    Stopped         2
>   PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
>   PS C:\Users\brian> wsl --list -v
>   NAME            STATE           VERSION
> * Ubuntu-18.04    Running         1
>   PS C:\Users\brian>wsl
>   brian@brian_pc:~$

Sorry, but this is absolutely no workaround. There are reasons why we use v2 and not v1.

**Comment by bjd145** [comment]
> > i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved.
> > PS C:\Users\brian> wsl --list -v
> > NAME STATE VERSION
> > 
> > * Ubuntu-18.04    Stopped         2
> >   PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
> >   PS C:\Users\brian> wsl --list -v
> >   NAME            STATE           VERSION
> > * Ubuntu-18.04    Running         1
> >   PS C:\Users\brian>wsl
> >   brian@brian_pc:~$
> 
> Sorry, but this is absolutely no workaround. There are obvious reasons why we use v2 and not v1.

I'm not sure why the down vote on my comment. I should made myself clear in that post I'm speaking for myself and not in any official capacity from Microsoft.  I was trying to be helpful to the community to folks who need to have their wsl distros up and functional. 

v2 is vastly better than v1. We all love v2 but if it's a matter between up but limping along versus being down then I'll take being up while the issue is being worked on.

**Comment by ad-on-is** [comment]
> Thanks, we are debugging this internally - trying to figure out why only some users are hitting this.

There's really some strategic solution needed here regarding Windows 10 Updates and especially WSL. 

What I mean is, yes, we opted for the Fast Ring and we know what risks we are taking as Beta-Testers. But WSL2 has become a crucial part of our work environment.
Iin fact, so crucial, that we do rely more on a stable working WSL than we do on a stable Windows itself. We spend more time in WSL, Terminal and VSCode than we do with Explorer or [insert any other fancy-shmency Windows feature here]. 

So please, for the sake of our work and time, can you guys at MS be more careful with such updates. We really love WSL, and are looking forward to all the new features you're going to provide us with. And most importantly, we are truly thankful for that.🤘

**Comment by ad-on-is** [comment]
> > > i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved.
> > > PS C:\Users\brian> wsl --list -v
> > > NAME STATE VERSION
> > > 
> > > * Ubuntu-18.04    Stopped         2
> > >   PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
> > >   PS C:\Users\brian> wsl --list -v
> > >   NAME            STATE           VERSION
> > > * Ubuntu-18.04    Running         1
> > >   PS C:\Users\brian>wsl
> > >   brian@brian_pc:~$
> > 
> > 
> > Sorry, but this is absolutely no workaround. There are obvious reasons why we use v2 and not v1.
> 
> I'm not sure why the down vote on my comment. I should made myself clear in that post I'm speaking for myself and not in any official capacity from Microsoft. I was trying to be helpful to the community to folks who need to have their wsl distros up and functional.
> 
> v2 is vastly better than v1. We all love v2 but if it's a matter between up but limping along versus being down then I'll take being up while the issue is being worked on.

Since the title already says **"WSL2-Error..."** there's really no need to suggest a workaround for converting to v1. 

It's not about which one is better. There are features that v1 can do but v2 can't, and vice versa. Depending on these features and based on our needs, we have distros running as v1 or v2, simultaneously.

**Comment by therealkenc** [comment]
For folks who can get into their WSL 2 shell om 19555, but are experiencing the "VS Code Remote - WSL" problem mentioned [above](https://github.com/microsoft/WSL/issues/4860#issuecomment-580758502) (looks like screencap), you can work-around by using ["VS Code Remote SSH"](https://code.visualstudio.com/docs/remote/ssh).

![image](https://user-images.githubusercontent.com/7049700/73571044-85a54880-4422-11ea-91d7-8071783025e3.png)

**Comment by nunix** [comment]
quick questions: 
do you use a custom kernel?
what does your .wslconfig looks like?
your distro 19-10 is custom tight? do you face the same issue with a store one?

I know the team is debugging, but I got some time this error when playing around with the kernel 🤔

**Comment by ad-on-is** [comment]
> quick questions:
> do you use a custom kernel?
> what does your .wslconfig looks like?
> your distro 19-10 is custom tight? do you face the same issue with a store one?
> 
> I know the team is debugging, but I got some time this error when playing around with the kernel

I've already reverted Windows back to the previous version. It's the official WSL-Ubuntu downloaded from the Ubuntu-Website, nothing custom and also no custom kernel here.

**Comment by JMBBunch** [comment]
I updated to Windows 10 Pro Insider Preview 19555.rs_prerelease.200127-0900 this morning. I first noticed the above issues with WSL when opening VSCode after the update (same problem with VSCode x64 v1.41.0 and 1.41.1 and I also did update its Remote - WSL extension to 0.41.7). 

I have nothing of importance in my WSL so upon experiencing these symptoms (and trying the SSL-related fix mentioned above which did not help), I figured I'd get rid of the Ubuntu 18.04 LTS distro I have installed and have a fresh go at it.  

Upon opening the WSL window to finish the installation, I get this:  
![image](https://user-images.githubusercontent.com/60524865/73573565-e3d92800-4430-11ea-8cec-f78fd2838651.png)

In case I didn't paste that in right, the upshot is the new installation prompts for a new default user account, as expected. I created one ("ubuntu") and it said the password was updated successfully, but then it came back with a connection attempt failure, which looks a lot like the same problem in my previously-working setup even though that error was within VSCode's terminal window and this one is in the WSL command window:  "A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond." 

As you can see in the screenshot, I thought I'd try a different name just in case, but the same error came up and came back to another new UNIX username prompt. 

In case it matters, after I updated Win10, I also installed (for the first time) the new cross-platform Powershell (6.2.4 x64). It was then I discovered VSCode couldn't open WSL.

**Comment by kobenauf** [comment]
Similar message for me.  I upgraded to build 19555 and now WSL 2 is totally dead:
`A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`
Interestingly, the icon on my taskbar that launches Ubuntu is also wrong now.  It used to be a little picture of a command prompt, but now it's a white blank page.
![image](https://user-images.githubusercontent.com/3782555/73577807-d4300280-4475-11ea-913c-505ed044dc62.png)

**Comment by benhillis** [comment]
@kobenauf - The roll back feature works very well and is actually quite quick.

**Comment by benhillis** [comment]
We were able to debug what the issue is here and are pushing the fix.  This issue triggers if you have a somewhat a long %PATH%, so one potential workaround (until a new build comes out with the fix) would be to do some "spring cleaning" on your %PATH% variable to make it shorter.

**Comment by ad-on-is** [comment]
> We were able to debug what the issue is here and are pushing the fix. This issue triggers if you have a somewhat a long %PATH%, so one potential workaround (until a new build comes out with the fix) would be to do some "spring cleaning" on your %PATH% variable to make it shorter.

By %PATH% variable you mean the user defined one within Windows?

**Comment by benhillis** [comment]
@ad-on-is - Correct.

**Comment by kobenauf** [comment]
@benhillis thanks for saying that about rolling back.  I've never actually done that before and was afraid it would corrupt my system.  (That happened to many, many, ... many years ago.)  And you're right, it was pretty quick and it fixed the problem.  Thanks!

**Comment by JMBBunch** [comment]
Reporting back:  I cleaned up my PATH (there were terribly long Intel Management Engine related PATHs for both /Program Files/ and /Program Files (x86)/ so I removed the x86 ones) and rebooted. I just put Ubuntu back (from the Windows Store) and VOILA! It started fine. 

However, VSCode is still showing the same error.  So to get it all clean, I uninstalled/reinstalled VSCode again, and removed and reinstalled the Remote - WSL extension; no help. I also pruned my path some more and reset it from the command line (and exited and re-entered to make sure the new path was retained) and tried again - same thing.  Here's VSCode's Terminal output after trying a Remote-WSL: New Window command:
![image](https://user-images.githubusercontent.com/60524865/73583296-a1bedf00-444e-11ea-9f7d-273f9ee33dbb.png)

**Comment by laoshancun** [comment]
same issue on 19555.1001

**Comment by benszymkow** [comment]
> We were able to debug what the issue is here and are pushing the fix. This issue triggers if you have a somewhat a long %PATH%, so one potential workaround (until a new build comes out with the fix) would be to do some "spring cleaning" on your %PATH% variable to make it shorter.

Is the new build going to come out in next few days or is it going to be a week+? Just want to figure out whether to roll back or just wait

**Comment by philwilliammee** [comment]
reduced my windows $PATH length to 458 characters, WSL2 path length 637 and wsl2 started but vscode cant connect to server. same error as others https://github.com/microsoft/WSL/issues/4860#issuecomment-580967091 adding startup output below as text.
```bash
[2020-02-01 12:06:06.597] Resolving wsl+Ubuntu, resolveAttempt: 1
[2020-02-01 12:06:06.648] Starting VS Code Server inside WSL (Ubuntu)
[2020-02-01 12:06:06.648] Extension version: 0.41.7, Windows build: 19555. Multi distro support: available. WSL path support: enabled
[2020-02-01 12:06:06.795] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2020-02-01 12:06:16.880] Unable to detect if server is already installed: Error: Command failed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2020-02-01 12:06:16.880] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
[2020-02-01 12:06:16.880] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 26076a4de974ead31f97692a0d32f90d735645c0 stable .vscode-server 0  ' in c:\Users\USER\.vscode\extensions\ms-vscode-remote.remote-wsl-0.41.7
[2020-02-01 12:06:26.953] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
[2020-02-01 12:06:26.953] VS Code Server for WSL closed unexpectedly.
[2020-02-01 12:06:26.953] For help with startup problems, go to
[2020-02-01 12:06:26.953] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
[2020-02-01 12:06:26.965] WSL Daemon exited with code 0
```

**Comment by FabienD** [comment]
> reduced my windows $PATH length to 458 characters, WSL2 path length 637 and wsl2 started but vscode cant connect to server. same error as others [#4860 (comment)](https://github.com/microsoft/WSL/issues/4860#issuecomment-580967091) adding startup output below as text.
> 
> ```shell
> [2020-02-01 12:06:06.597] Resolving wsl+Ubuntu, resolveAttempt: 1
> [2020-02-01 12:06:06.648] Starting VS Code Server inside WSL (Ubuntu)
> [2020-02-01 12:06:06.648] Extension version: 0.41.7, Windows build: 19555. Multi distro support: available. WSL path support: enabled
> [2020-02-01 12:06:06.795] Probing if server is already installed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
> [2020-02-01 12:06:16.880] Unable to detect if server is already installed: Error: Command failed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
> [2020-02-01 12:06:16.880] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> [2020-02-01 12:06:16.880] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 26076a4de974ead31f97692a0d32f90d735645c0 stable .vscode-server 0  ' in c:\Users\USER\.vscode\extensions\ms-vscode-remote.remote-wsl-0.41.7
> [2020-02-01 12:06:26.953] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> [2020-02-01 12:06:26.953] VS Code Server for WSL closed unexpectedly.
> [2020-02-01 12:06:26.953] For help with startup problems, go to
> [2020-02-01 12:06:26.953] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
> [2020-02-01 12:06:26.965] WSL Daemon exited with code 0
> ```

Have you checked that SSH server runs on the host ? In my case, ssh-server wouldn't start after upgrade (few keys were missing). I don't know if it happened after Windows or Ubuntu udpates, I've done both in the same time. 
After uninstall openssh-server (purge), re-install it, remove all .vcode dirs, on host and window (so re-install plugins), all works fine again.

**Comment by alekseymvt** [comment]
> I just put Ubuntu back (**from the Windows Store**) and VOILA! It started fine.

Check, please, Wsl version: `wsl -l -v`
Do you change2 version to 2? `wsl --set-version`

**Comment by kobenauf** [comment]
I had rolled back yesterday to 19551, which fixed the problem.  At the same time I cleaned up my Windows paths as suggested above, and this morning I see Windows did the auto-update to 19555 again.  This time I can indeed launch WSL 2 again.  However, I have the problem @philwilliammee posted above, that VSC won't launch.  Same error as him.  As @FabienD suggested, I checked if ssh server was running, and it was not.  When I try to start it, I see this error:
```
sudo service ssh start
* Starting OpenBSD Secure Shell server sshd
Could not load host key: /etc/ssh/ssh_host_rsa_key
Could not load host key: /etc/ssh/ssh_host_ecdsa_key
Could not load host key: /etc/ssh/ssh_host_ed25519_key
```
I ran `sudo service ssh start` a second time, and it started without errors.
```
sudo service ssh status
* sshd is running
```
I should point out, I've never had to do this on previous upgrades, so something is a bit different this time.

However after starting the ssh server, VSC still has the same error when it opens.
```
| ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2020-02-01 14:56:02.345] Unable to detect if server is already installed: Error: Command failed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
[2020-02-01 14:56:02.345] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
[2020-02-01 14:56:02.345] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 26076a4de974ead31f97692a0d32f90d735645c0 stable .vscode-server 0  ' in c:\Users\Kip Obenauf\.vscode\extensions\ms-vscode-remote.remote-wsl-0.41.7
[2020-02-01 14:56:12.422] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
[2020-02-01 14:56:12.422] VS Code Server for WSL closed unexpectedly.
[2020-02-01 14:56:12.422] For help with startup problems, go to
[2020-02-01 14:56:12.422] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
```
On Windows, I deleted `C:\Users\Kip Obenauf\.vscode`.  When I started `code .` again, it told me the Remote extension wasn't installed, so it installed it.  I restarted `code .` afterwards, but I still get the same error.

**Comment by FabienD** [comment]
> I had rolled back yesterday to 19551, which fixed the problem. At the same time I cleaned up my Windows paths as suggested above, and this morning I see Windows did the auto-update to 19555 again. This time I can indeed launch WSL 2 again.
> 
> However, I have the problem @philwilliammee posted above, that VSC won't launch. Same error as him. As @FabienD suggested, I checked if ssh server was running, and it was not. When I try to start it, I see this error:
> 
> ```
> sudo service ssh start
> * Starting OpenBSD Secure Shell server sshd
> Could not load host key: /etc/ssh/ssh_host_rsa_key
> Could not load host key: /etc/ssh/ssh_host_ecdsa_key
> Could not load host key: /etc/ssh/ssh_host_ed25519_key
> ```
> 
> I ran `sudo service ssh start` a second time, and it started without errors.
> 
> ```
> sudo service ssh status
> * sshd is running
> ```
> 
> I should point out, I've never had to do this on previous upgrades, so something is a bit different this time.
> 
> However after starting the ssh server, VSC still has the same error when it opens.
> 
> ```
> | ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
> [2020-02-01 14:56:02.345] Unable to detect if server is already installed: Error: Command failed: C:\WINDOWS\System32\wsl.exe -d Ubuntu -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
> [2020-02-01 14:56:02.345] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> [2020-02-01 14:56:02.345] Launching C:\WINDOWS\System32\wsl.exe -d Ubuntu sh -c '"$VSCODE_WSL_EXT_LOCATION/scripts/wslServer.sh" 26076a4de974ead31f97692a0d32f90d735645c0 stable .vscode-server 0  ' in c:\Users\Kip Obenauf\.vscode\extensions\ms-vscode-remote.remote-wsl-0.41.7
> [2020-02-01 14:56:12.422] A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> [2020-02-01 14:56:12.422] VS Code Server for WSL closed unexpectedly.
> [2020-02-01 14:56:12.422] For help with startup problems, go to
> [2020-02-01 14:56:12.422] https://code.visualstudio.com/docs/remote/troubleshooting#_wsl-tips
> ```
> 
> On Windows, I deleted `C:\Users\Kip Obenauf\.vscode`. When I started `code .` again, it told me the Remote extension wasn't installed, so it installed it. I restarted `code .` afterwards, but I still get the same error.

Hi, have checked ssh-server status after error ? Have you try to reinstall it ?

**Comment by tforster** [comment]
Rolling back worked for me and it was less time than it took to make a cup of tea. Kudos to all the MS engineers that make the Insiders program reliable enough that the rest of us can be at the leading edge, while still having a dependable safety net for these occasional hiccups.

**Comment by kobenauf** [comment]
@FabienD  I reinstalled openssh-server, but I still get the same error.  At this point I am also just rolling back to 19551 so I can do work.

**Comment by PhilDim1** [comment]
Same issue, here.

**Comment by dancosta** [comment]
> > > > i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved.
> > > > PS C:\Users\brian> wsl --list -v
> > > > NAME STATE VERSION
> > > > 
> > > > * Ubuntu-18.04    Stopped         2
> > > >   PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
> > > >   PS C:\Users\brian> wsl --list -v
> > > >   NAME            STATE           VERSION
> > > > * Ubuntu-18.04    Running         1
> > > >   PS C:\Users\brian>wsl
> > > >   brian@brian_pc:~$
> > > 
> > > 
> > > Sorry, but this is absolutely no workaround. There are obvious reasons why we use v2 and not v1.
> > 
> > 
> > I'm not sure why the down vote on my comment. I should made myself clear in that post I'm speaking for myself and not in any official capacity from Microsoft. I was trying to be helpful to the community to folks who need to have their wsl distros up and functional.
> > v2 is vastly better than v1. We all love v2 but if it's a matter between up but limping along versus being down then I'll take being up while the issue is being worked on.
> 
> Since the title already says **"WSL2-Error..."** there's really no need to suggest a workaround for converting to v1.
> 
> It's not about which one is better. There are features that v1 can do but v2 can't, and vice versa. Depending on these features and based on our needs, we have distros running as v1 or v2, simultaneously.

I personally think that @bjd145 comment was a valid idea. It may not be a option to anyone using WSL2 but may be a option. In my case I had just 2 options downgrading my Ubuntu or Rollingback the Windows update! I chose the latter option because I wanted to see how it is to rollback windows update and I was encouraged to do so because some people here reported they successfully did it. 

And @bjd145 was kind enough to give the complete set of commands to use his workaround to the problem. 

Thanks everyone here and MS to be doing such a great work. It is good to know MS is already doing a new build to fix it @benhillis =)

**Comment by JMBBunch** [comment]
@FabienD I did have the SSH server issues. It was not running, and when I started it the key problems appeared. So I did purge and reinstall, and then SSH server started up without errors. It now comes up as Running when I open WSL. This did not solve the VSCode issue, unfortunately.

Incidentally, I installed both Ubuntu and Ubuntu-18.04 from the Store. I ensured both are set to version 2.  VSCode is not coming up for either of them.  Same messages.  (Same SSH server issues and fix attempts also.)

Then, I tried @FabienD 's suggestion to remove the .vscode folder from the User profile, start VSCode from the WSL command line with "code .", and install the Remote extension when prompted, but I am still getting the same errors. 

Thought I'd dig a little more. The terminal output in VSCode shows a line that starts, "Probing if server is already installed", and then issues the command:

C:\WINDOWS\System32\wsl.exe -d Ubuntu-18.04 -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"

That is followed by the error message, "Unable to detect if server is already installed", so I thought I'd test the above command line in the various Windows terminals. All of those successfully return: "found"

PATH in Windows is 534 characters.
PATH in WSL (Ubuntu 18.04) is 803 characters.

Does anyone know whether these paths are still too long?

**Comment by ad-on-is** [comment]
> > > > > i was hitting the same issue with my v2 distro then i successfully converted it back to v1. I can now access the distro. Took about an hour to convert down but at least it's a viable workaround until the issue is resolved.
> > > > > PS C:\Users\brian> wsl --list -v
> > > > > NAME STATE VERSION
> > > > > 
> > > > > * Ubuntu-18.04    Stopped         2
> > > > >   PS C:\Users\brian> wsl --set-version Ubuntu-18.04 1
> > > > >   PS C:\Users\brian> wsl --list -v
> > > > >   NAME            STATE           VERSION
> > > > > * Ubuntu-18.04    Running         1
> > > > >   PS C:\Users\brian>wsl
> > > > >   brian@brian_pc:~$
> > > > 
> > > > 
> > > > Sorry, but this is absolutely no workaround. There are obvious reasons why we use v2 and not v1.
> > > 
> > > 
> > > I'm not sure why the down vote on my comment. I should made myself clear in that post I'm speaking for myself and not in any official capacity from Microsoft. I was trying to be helpful to the community to folks who need to have their wsl distros up and functional.
> > > v2 is vastly better than v1. We all love v2 but if it's a matter between up but limping along versus being down then I'll take being up while the issue is being worked on.
> > 
> > 
> > Since the title already says **"WSL2-Error..."** there's really no need to suggest a workaround for converting to v1.
> > It's not about which one is better. There are features that v1 can do but v2 can't, and vice versa. Depending on these features and based on our needs, we have distros running as v1 or v2, simultaneously.
> 
> I personally think that @bjd145 comment was a valid idea. It may not be a option to everyone using WSL2 but may be an option. In my case I had just 2 options: downgrading my Ubuntu or Rollingback the Windows update! I chose the latter option because I wanted to see how it is to rollback windows update, and I was encouraged to do so because some people here reported they successfully did it.
> 
> And @bjd145 was kind enough to give the complete set of commands to use his workaround to the problem.
> 
> Thanks everyone here and MS to be doing such a great work. It is good to know MS is already doing a new build to fix it @benhillis =)

If you use WSL to just play around with Linux, and want to try all these geeky stuff everyone is talking about, then sure, there are many options. 

But, if you work on client projects, where time and reliability play a significant role, you definitely don't want to waste time experimenting with nonsense workarounds. Especially not when you have projects (sourcecode) stored inside WSL (in a vhdx-file) which could eventually end up in data corruption or even data loss. 

You just want to get back to a working environment as soon as possible. 
Thankfully restoring Windows nowadays is just a matter of minutes and not hours.

**Comment by ad-on-is** [comment]
> @FabienD I did have the SSH server issues. It was not running, and when I started it the key problems appeared. So I did purge and reinstall, and then SSH server started up without errors. It now comes up as Running when I open WSL. This did not solve the VSCode issue, unfortunately.
> 
> Incidentally, I installed both Ubuntu and Ubuntu-18.04 from the Store. I ensured both are set to version 2. VSCode is not coming up for either of them. Same messages. (Same SSH server issues and fix attempts also.)
> 
> Then, I tried @FabienD 's suggestion to remove the .vscode folder from the User profile, start VSCode from the WSL command line with "code .", and install the Remote extension when prompted, but I am still getting the same errors.
> 
> Thought I'd dig a little more. The terminal output in VSCode shows a line that starts, "Probing if server is already installed", and then issues the command:
> 
> C:\WINDOWS\System32\wsl.exe -d Ubuntu-18.04 -e sh -c "[ -d ~/.vscode-server/bin/26076a4de974ead31f97692a0d32f90d735645c0 ] && printf found || ([ -f /etc/alpine-release ] && printf alpine-; uname -m)"
> 
> That is followed by the error message, "Unable to detect if server is already installed", so I thought I'd test the above command line in the various Windows terminals. All of those successfully return: "found"
> 
> PATH in Windows is 534 characters.
> PATH in WSL (Ubuntu 18.04) is 803 characters.
> 
> Does anyone know whether these paths are still too long?

Even though I rolled back, I checked my Windows PATH out of curiosity, and I wouldn't consider them to be long, since  therer are only 4 lines, AppData, VScode, Flutter-sdk and Dashlane paths, which are crucial for me.

**Comment by PhilDim1** [comment]
> Even though I rolled back, I checked my Windows PATH out of curiosity, and I wouldn't consider them to be long, since therer are only 4 lines, AppData, VScode, Flutter-sdk and Dashlane paths, which are crucial for me.

Mine was nearly that short. Nothing excessive, not in length or sheer number.

**Comment by vanguard737** [comment]
Regarding $PATH length, just in case this helps anybody:  I didn't want to make any persistent changes to my system $PATH.  So I just did some temporary $PATH-pruning within Powershell, with decent results.

```
# My default $PATH is pretty long...
PS C:\WINDOWS\system32> $Env:Path.length
1651
# ...and at that length, I can't login:
PS C:\WINDOWS\system32> wsl
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
# ...So, I'll temporarily make the path as short as humanly possible:
PS C:\WINDOWS\system32> $Env:Path = "C:\Windows\System32;C:\Windows";
# ...which is pretty short:
PS C:\WINDOWS\system32> $Env:Path.length
30
# ...And now I can login without issue.  Victory!
PS C:\WINDOWS\system32> wsl
root@DESKTOP-UVCU1I3:/mnt/c/WINDOWS/system32#
```

**Comment by jbudz** [comment]
Thanks for the workarounds everyone.  I didn't feel like fiddling too much so gave the rollback approach a try.  I went from 19555 to 19551 - it took about 15 minutes.  generic disclaimer about removing updates

Start -> Settings -> Update and Security -> Recovery -> Go back to the previous version of Windows 10

**Comment by PhilDim1** [comment]
> Regarding $PATH length, just in case this helps anybody: I didn't want to make any persistent changes to my system $PATH. So I just did some temporary $PATH-pruning within Powershell, with decent results.
> 
> ```
> # My default $PATH is pretty long...
> PS C:\WINDOWS\system32> $Env:Path.length
> 1651
> # ...and at that length, I can't login:
> PS C:\WINDOWS\system32> wsl
> A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> # ...So, I'll temporarily make the path as short as humanly possible:
> PS C:\WINDOWS\system32> $Env:Path = "C:\Windows\System32;C:\Windows";
> # ...which is pretty short:
> PS C:\WINDOWS\system32> $Env:Path.length
> 30
> # ...And now I can login without issue.  Victory!
> PS C:\WINDOWS\system32> wsl
> root@DESKTOP-UVCU1I3:/mnt/c/WINDOWS/system32#
> ```

Thanks, this actually worked for me. I had no idea my $PATH was that large. I cut mine down to ~700, it works. Weird issue

**Comment by alexign** [comment]
same boat here.
 just truncate Windows %PATH% from lengthy string to some:
C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\HashiCorp\Vagrant\bin;C:\Users\username\AppData\Local\Microsoft\WindowsApps;C:\Users\username\AppData\Local\Programs\Microsoft VS Code\bin

and now code  can open wsl2 dir's

**Comment by stefanjarina** [comment]
So rollback it is as truncating PATH is not an option if one have tons and tons of software that needs to be there. This PATH length limit in windows is just terrible, especially when software such as intel or razer pollute it with several really long paths.

**Comment by BabyDino** [comment]
I just deleted my previous Windows installation files before starting vscode. Not my smartest move.

I agree with @ad-on-is that using the fast ring is a risk but all latest builds seems to have serious issues which makes it unusable as a developer. Truncating %PATH% seems to do the trick for now.

FWIW: my $Env.Path.length is 217 chars which seems to work.

**Comment by uniibu** [comment]
Like the other above, shortening your `%PATH%` seems to work. The length of my %PATH% was 1800+ and now down to 1288 and wsl now works fine. 

To shorten my PATH, I removed duplicates variables. You could edit your PATH by going to System Properties -> Advance -> Environment Variables. And under System Variables, look for Path, then edit.

I found that I had duplicate entries like `C:\WINDOWS`, `C:\Windows`, `%SystemRoot%`, `C:\WINDOWS\System32`, `C:\Windows\System32`, `%SystemRoot%\System32` etc.. So i removed the `%SystemRoot%` and `C:\WINDOWS`(capitals) duplicate entries, and this significantly reduced my PATH variable.

**Comment by wagneripjr** [comment]
> We were able to debug what the issue is here and are pushing the fix. This issue triggers if you have a somewhat a long %PATH%, so one potential workaround (until a new build comes out with the fix) would be to do some "spring cleaning" on your %PATH% variable to make it shorter.

Thank!, I've shortened the path (removed SQL Server entries 😄 ) and WSL is working again

**Comment by pellea** [comment]
Shortening the $PATH is working to start WSL but I still cannot start my VSCode env. This workaround is useless for me. 

I revert back to the previous build version of Windows.

**Comment by sc0ttwad3** [comment]
@uniibu I frequently run into those duplicated C:\Windows, C:\Windows\System32, ...and the rest, entries in the system path. Over the course of Windows 10 (since the first available builds years ago), I think I have removed those duplicates at least dozen or more times. And still haven't figured out when/what re-adds the unecessarily.

**Comment by gaffneyd4** [comment]
Shortened my path down to 1048, appears to be working fine now.
Thanks to everyone for all of the tips, I was honestly surprised to see so much activity on an issue created 3 days ago! Its cool to see that the community is adopting these WSL features as quickly as they're released! I don't say this often, but keep up the great work MS! 🙌

**Comment by lengthmin** [comment]
I have reverted to 19551.
But I found a reg value that seems to help:
```powershell
Set-ItemProperty 'HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem' -Name 'LongPathsEnabled' -Value 1
```

from: https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/

**I haven't tried, just to share.**

**Comment by skuzzer** [comment]
> I have reverted to 19551.
> But I found a reg value that seems to help:
> 
> ```powershell
> Set-ItemProperty 'HKLM:\SYSTEM\CurrentControlSet\Control\FileSystem' -Name 'LongPathsEnabled' -Value 1
> ```
> 
> from: https://www.howtogeek.com/266621/how-to-make-windows-10-accept-file-paths-over-260-characters/
> 
> **I haven't tried, just to share.**

This doesn't work.

I've managed to shorten my path to 956 chars and WSL2 now launches but the VS code extension still fails

**Comment by NaiveTomcat** [comment]
I might have figured out a workaround by shutting down and restart wsl.

Only workable in Powershell.

see #4866

**Comment by NaiveTomcat** [comment]
I might have figured out a workaround by shutting down and restart wsl.

Only workable in Powershell.

see #4866

**Comment by NaiveTomcat** [comment]
> I might have figured out a workaround by shutting down and restart wsl.
> 
> Only workable in Powershell.
> 
> see #4866

And my %PATH% length is 1054 chracters

**Comment by balcsida** [comment]
Hi,

I just also wanted to confirm that reducing the path variable length works:
I had a PATH with the length of 1330 character and educed it to 1005 characters and WSL works again.

**Comment by Xepol** [comment]
Roll back was the only solution here.  Reducing path <1000 chars (and even rebooting) had no effect.  Thankfully I backed up my .vhdx before I rolled back here, because rolling back destroyed all my wsl installs.

So, shut down the lsxx service, copy out the .vhdx files for anything that matters, rollback.  Reinstall any required WSL installations, create your users, then shut down lsxx service again, copy back your .vhdx files and you should be back exactly where you were.

Not quite 15 minutes because of the copying (my .vhdx is quite large), but still better than the wall of hurt this bug (whatever it is) causes.

**Comment by JMBBunch** [comment]
It appears the remaining problem (VS Code with WSL) has been narrowed down to the Remote - WSL extension (at least that is what a bot reported), and this is the link to the issue as reported there:    
https://github.com/microsoft/vscode-remote-release/issues/2272

**Comment by JMBBunch** [comment]
Here's a workaround for those who don't want to roll back Win10:

First, shorten the PATH enough to get WSL to run.
Then, shorten it some more (probably a lot) to get VS Code to run. 

I could not get my Windows PATH pruned enough to risk breaking what I need in Windows. Instead, reset the PATH for the session by starting up WSL within Powershell:

1. Open Powershell.
2. $Env:PATH="C:\Users\username\AppData\Local\Microsoft\WindowsApps;C:\Users\username\AppData\Local\Programs\Microsoft VS Code\bin;"
3. C:\windows\system32\wsl.exe
The prompt changes to WSL, sitting at /mnt/c/WINDOWS/system32
4. Change to the WSL folder you want VS Code to open:  cd ~
5. Run VS Code:  code .
Woo hoo! We have WSL folders in VS Code again!

Hope this helps someone!

**Comment by therealkenc** [comment]
Equivalent in `cmd.exe` is [`set`](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/set_1)`PATH=<min path you need for interop>`. [Contrast [`setx`](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/setx).]

**Comment by chensjlv** [comment]
To help people, I think you still need vscode path in your $PATH for the vscode server to launch. Here are my configs

User's Path

```
C:\Users\{youusername}\AppData\Local\Programs\Microsoft VS Code\bin;
```

System's path
```
C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\PowerShell\6\;
```

**Comment by LiamKarlMitchell** [comment]
So how to fix this?

![image](https://user-images.githubusercontent.com/1968109/73713845-b3b4c080-4773-11ea-90ce-26c958a9afb2.png)

**Comment by aeschli** [comment]
I pushed new versions of the Remote - WSL extension (0.41.9, 0.42.3) with a workaround to overcome the PATH length issues (the extension uses a empty PATH when internally calling wsl.exe).

**Comment by keevan** [comment]
Just wanted to link this blog post I found useful regarding the suggested 'spring clean'
https://peter.hahndorf.eu/blog/BloatedPathVariable.html

Essentially the command to list out all the paths is:
```
($env:path -split ";") | ? {if (Test-Path ([System.Environment]::ExpandEnvironmentVariables("$_"))){Write-Host $_}else{Write-Host $_ -f red}}
```

The quick fix to get it loading (by overriding the PATH) worked for me, but I'm currently seeing if I can clean up my PATH length enough to get it to work. 

I started from a length of 2000+, _still not working at 1291_
```
$Env:Path.length
```

**Comment by diddledan** [comment]
There's something more than simple length of the `PATH` variable, because I'm down well below where others have managed to get it working at 536 characters yet it still isn't able to launch here with the message as reported by others:

```plain
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

My `PATH` variable now contains:

```plain
C:\Users\yabea> $env:path -split ";"
C:\Program Files (x86)\Common Files\Oracle\Java\javapath
C:\Program Files\PowerShell\6\
C:\HaxeToolkit\haxe
C:\HaxeToolkit\neko
C:\Program Files (x86)\GnuPG\bin
C:\WINDOWS\system32
C:\WINDOWS
C:\WINDOWS\System32\Wbem
C:\WINDOWS\System32\WindowsPowerShell\v1.0\
C:\WINDOWS\System32\OpenSSH\
C:\Program Files\Git\cmd
C:\Program Files\dotnet\
C:\Program Files\Docker\Docker\resources\bin
C:\ProgramData\DockerDesktop\version-bin
C:\Users\yabea\AppData\Local\Programs\Microsoft VS Code\bin
C:\Users\yabea\AppData\Local\Microsoft\WindowsApps
```

**Comment by NaiveTomcat** [comment]
> There's something more than simple length of the `PATH` variable, because I'm down well below where others have managed to get it working at 536 characters yet it still isn't able to launch here with the message as reported by others:
> 
> ```
> A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> ```
> 
> My `PATH` variable now contains:
> 
> ```
> C:\Users\yabea> $env:path -split ";"
> C:\Program Files (x86)\Common Files\Oracle\Java\javapath
> C:\Program Files\PowerShell\6\
> C:\HaxeToolkit\haxe
> C:\HaxeToolkit\neko
> C:\Program Files (x86)\GnuPG\bin
> C:\WINDOWS\system32
> C:\WINDOWS
> C:\WINDOWS\System32\Wbem
> C:\WINDOWS\System32\WindowsPowerShell\v1.0\
> C:\WINDOWS\System32\OpenSSH\
> C:\Program Files\Git\cmd
> C:\Program Files\dotnet\
> C:\Program Files\Docker\Docker\resources\bin
> C:\ProgramData\DockerDesktop\version-bin
> C:\Users\yabea\AppData\Local\Programs\Microsoft VS Code\bin
> C:\Users\yabea\AppData\Local\Microsoft\WindowsApps
> ```

Try to shrink `PATH` variable to null by 
```
$env:PATH=""
```
It should work and should make no influence to other application.

**Comment by rmja** [comment]
Clearing the path does not work for me:

```
> $env:PATH=""
> C:\Windows\System32\wsl.exe -d docker-desktop
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
````

**Comment by rishdas** [comment]
> Thanks, we are debugging this internally - trying to figure out why only some users are hitting this.

Thanks I am facing this issue too

**Comment by rishdas** [comment]
I am facing the same issue funny though I am not facing this issue on Windows Installation with a VM
Anyways switching to WSL (1) for now. Please update once we have solution

**Comment by benhillis** [comment]
@rishdas - The best thing to do would be to roll back to the previous build until a new Fast build is out with the fix.

**Comment by gianugo** [comment]
FYI - my path is 602 characters and I'm still getting the same error. The one (possible and minor) difference: I upgraded my distro to WSL2 after upgrading to 19555. Since then it didn't start anymore, and no shortening of %PATH% seems to be helping. I'm on corpnet, happy to provide access to my machine if it could help debugging.

**Comment by 24githubs** [comment]
I recently had to reinstall window  ,
The only thing that finally worked for me!!

version 2004 (OS Build 19555,1001)

**Comment by pbellusci** [comment]
Same Problem here, I really can't believe how you play with our environments. Don't you have a QA department to test all of this things? I'm contractor, who will pay for the hours that I'll have to lose to fix this?.

C'mon, be professional guys, for once in life.

**Comment by gel-mashup** [comment]
> Same Problem here, I really can't believe how you play with our environments. Don't you have a QA department to test all of this things? I'm contractor, who will pay for the hours that I'll have to lose to fix this?.
> 
> C'mon, be professional guys, for once in life.

we are in fast ring. these things are expected.. i suggest stop your update at 19551 or move to slow ring (slow ring version now supports wsl2)

**Comment by pauljmelia** [comment]
Also having this issue; reduced the length of my %PATH% below 500 and still couldn't get docker-desktop to start, reverted to previous build and paused updates for a week.

**Comment by rikkit** [comment]
> Same Problem here, I really can't believe how you play with our environments. Don't you have a QA department to test all of this things? I'm contractor, who will pay for the hours that I'll have to lose to fix this?.

By volunteering for fast ring you volunteered to join the QA department.

**Comment by jmar5439** [comment]
I discovered a workaround that is working in my system. I have Windows 10 PRO Versión	10.0.19041 compilación 19041. I 've installed WSL 2 and  configured Docker to work with wsl2.
Seems like docker is interfering some way and some processes are not managed correctly.
In order to start WSL 2 , I turned off docker, I disabled "Docker auto start" and "Windows Fast start up".

```
net stop docker
 net stop com.docker.service
```

**Comment by MWaser** [comment]
This really sucks for me as I've just joined the Insider program and don 't have a previous version to roll back to.  What can people like me do?

**Comment by pbellusci** [comment]
> > Same Problem here, I really can't believe how you play with our environments. Don't you have a QA department to test all of this things? I'm contractor, who will pay for the hours that I'll have to lose to fix this?.
> 
> By volunteering for fast ring you volunteered to join the QA department.

No way.
Rule #1 of software development... TEST things before push anything. We're in 2020, we have a tons of tools for testing.

This is not serious.

**Comment by TeamDman** [comment]
The insiders program _is_ the test, for before features are pushed to the masses.

I'd argue that it's irresponsible/unprofessional to depend solely on the stability of insiders to do your job, especially when virtual box + remote-ssh in vscode works just as well as wsl remoting does.

That's not to say that this issue isn't an inconvenience, but the entitlement being displayed here by many is laughable.

**Comment by cmeiklejohn** [comment]
Just ran into the problem, was able to revert to 19551 and get it working again.

**Comment by ad-on-is** [comment]
> The insiders program _is_ part of the testing, for before features are pushed to the masses.
> 
> I'd argue that it's irresponsible/unprofessional to depend solely on the stability of insiders to do your job, especially when virtual box + remote-ssh in vscode works just as well as wsl remoting does.
> 
> That's not to say that this issue isn't an inconvenience, but the entitlement being displayed here by many is laughable.

Well, after WSL2 was realesed, users tested and loved it, in the way that it works right now. Since it's functionalities proved to be reliable and usable for daily work, we adopted it into our dev-environments. So we expect at least the core features to work, which already have been tested and verified. And we can at least expect that, can't we? It's not like we're whining and crying about some announced and untested nice-to-have features.

**Comment by benhillis** [comment]
Not to split hairs, but WSL2 has still not officially "released" - It is still only in Insider builds.

If you are on the Fast ring you will occasionally be impacted by bugs in WSL or other components. Having insiders test pre release versions of Windows is invaluable, and catches things like this that were missed during testing. Rolling back builds takes  minutes, it is no great hardship.

**Comment by benhillis** [comment]
Unfortunately 19559 does not have the fix.

**Comment by craigloewen-msft** [comment]
[Windows Insider build 19546](https://blogs.windows.com/windowsexperience/2020/02/12/announcing-windows-10-insider-preview-build-19564/) is out today and has a fix for this issue. Please give it a try and let us know if this is resolved for you!

**Comment by magneticflux-** [comment]
Build 19564 resolved the issue for me.

**Comment by n3dst4** [comment]
Fixed for me.

**Comment by ad-on-is** [comment]
Yep, 19564 fixed it!

**Comment by appurist** [comment]
Thanks so much. I can confirm that the fast ring update does address the issue. And I can confirm that WSL2 is available on the Windows Insiders slow ring.

Any idea when we might see this fix on the slow ring?

**Comment by philwilliammee** [comment]
fixed for me too thanks

**Comment by benhillis** [comment]
@appurist - This bug has never made it to the slow ring.

**Comment by appurist** [comment]
@benhillis Yes, that's why I was asking when it might hit the slow ring.  (Or a GA general release, for that matter.) For those people waiting for WSL2 who aren't prepared to run a Fast release.

**Comment by OpenWaygate** [comment]
I have fixed this problem in 19555.1001 by reducing `$Env:Path.length`, today my system update to 19559.1000 and I find a problem but I'm not sure whether it exists in 19555.1001.

The `bash` works only with Administrator.
![image](https://user-images.githubusercontent.com/27287694/74505354-e314bb80-4f31-11ea-81fc-7de92cd67dc4.png)

**Comment by Biswa96** [comment]
There is a new one, build 19564.

**Comment by SonicGD** [comment]
I'm on 19564 and still see this issue in windows terminal:

![image](https://user-images.githubusercontent.com/1222013/74506733-3976ff80-4f1c-11ea-898f-b0bcaedb50b8.png)

Only 1 of those 7 tabs opened normally, all 6 others with `[process exited with code 1]` error.

**Comment by sebagomez** [comment]
Don't if totally related but I'm also on 19564.1005 but I still have an error starting Docker

```
Docker.Core.Backend.BackendException:
Timed out waiting for the lifecycle-server to start.
This may be the result of a slow environment or there may be a real problem.
This timeout can be increased by editing settings.json, value 'lifecycleTimeoutSeconds'.
This file is usually in C:\Users\(your username)\AppData\Roaming\Docker
   at Docker.Core.Pipe.NamedPipeClient.<TrySendAsync>d__5.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Docker.Core.Pipe.NamedPipeClient.Send(String action, Object[] parameters)
   at Docker.Actions.DoRestartVM(Settings previousSettings)
   at Docker.Actions.<>c__DisplayClass34_0.<RestartVM>b__0()
   at Docker.ApiServices.TaskQueuing.TaskQueue.<>c__DisplayClass18_0.<.ctor>b__1()

  System.TimeoutException:
  Timed out waiting for the lifecycle-server to start.
This may be the result of a slow environment or there may be a real problem.
This timeout can be increased by editing settings.json, value 'lifecycleTimeoutSeconds'.
This file is usually in C:\Users\(your username)\AppData\Roaming\Docker
     at Docker.Backend.LifecycleClient.WaitForServerUp(TimeSpan pingTimeout)
     at Docker.Backend.ContainerEngine.WslEngine.DoStart(Settings settings, String daemonOptions, Credential credential)
     at Docker.Backend.ContainerEngine.WslEngine.Restart(Settings settings, String daemonOptions, Credential credential)
     at Docker.Backend.BackendNamedPipeServer.<Run>b__8_3(Object[] args)
     at Docker.Core.Pipe.NamedPipeServer.<>c__DisplayClass9_0.<Register>b__0(Object[] parameters)
     at Docker.Core.Pipe.NamedPipeServer.RunAction(String action, Object[] parameters)
  
```

**Comment by haydenrou** [comment]
I'm also on 19564 and am still struggling to open WSL #4892

**Comment by ShadowEO** [comment]
Same as @haydenrou, updated to 19564 and can't open at all, though in my case, I can't even get a WSL session to start after butchering the path variable to nothing now. Still getting "A connection attempt failed" with a blank PATH now.

**Comment by Coldrain1232008** [comment]
Same as above. I'm 19564.1005 with [process exited with code 1] error.

**Comment by LunchWithaLens** [comment]
This is working for me again now - 19564 Insider Preview.  I had trimmed down my $PATH in earlier builds with no success.  It did take a while to start though.

**Comment by haydenrou** [comment]
#4892 is still broken in 19569, any update?

**Comment by sebagomez** [comment]
I'm also on 19569 and the error I published before was still occurring (https://github.com/microsoft/WSL/issues/4860#issuecomment-586294413)

Fixed it by uninstalling and installing the Docker Client back again.

**Comment by tylerjkline** [comment]
> [Windows Insider build 19546](https://blogs.windows.com/windowsexperience/2020/02/12/announcing-windows-10-insider-preview-build-19564/) is out today and has a fix for this issue. Please give it a try and let us know if this is resolved for you!

Did NOT fix for me, was working fine until I shutdown WSL2 and tried to restart it for other things, I'm running build 19569.

**Comment by mafshin** [comment]
On Build 19559.1000 and WSL2 complains to start

**Comment by pkostelnik** [comment]
No Problem anymore with WSL2 since 19582

**Comment by willy1222** [comment]
> > [Windows Insider build 19546](https://blogs.windows.com/windowsexperience/2020/02/12/announcing-windows-10-insider-preview-build-19564/) is out today and has a fix for this issue. Please give it a try and let us know if this is resolved for you!
> 
> Did NOT fix for me, was working fine until I shutdown WSL2 and tried to restart it for other things, I'm running build 19569.

Same issue:
`PS C:\Users\willy> wsl`
`A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`

I'm running build 19564 and the following commands work for me.
`PS C:\Users\willy> wsl --shutdown`
`PS C:\Users\willy> wsl`

**Comment by Kenney-G** [comment]
Having this issue with Arch, Windows ver 19624.

**Comment by rs-13-11** [comment]
I am facing similar issue on ubuntu 20.04 wsl 2 especially after installing docker desktop

![image](https://user-images.githubusercontent.com/52852457/85353844-8ca53280-b54c-11ea-93ad-901988eac816.png)

**Comment by homandiy** [comment]
How can I make WSL automatically shut down if the error pops up again?

**Comment by therealkenc** [comment]
There are some open landing zones #5459 #5092 for the same error but with presumably unrelated cause. This issue, which could be triggered reliably with a long %path% in a narrow window between 19555  and 19564 came and went.

**Comment by arakno** [comment]
Clicking on left hand corner of VS Code and choosing to open new Window did it for me
![image](https://user-images.githubusercontent.com/2115768/94365107-26d5c300-00c6-11eb-96a2-867f64261730.png)

**Comment by rogatec** [comment]
@arakno nice catch!

Some strange behavior occurs when a vs code instance with a remoted connected project from wsl was opened.
After a reboot of the PC and reopening vs code, it tries to reopen the last opened folder/project, which fails.
In this situation the complete wsl vm is blocked `wsl --list -v` shows that the vm is stopped. VS Code could therefore also not establish a connection.

After @arakno 's hint of opening a new "clean" window from vs code with no open remote connection, suddenly the `wsl --list -v` command showed that the wsl vm is running.

Maybe the WSL and VS Code Team can investigate this strange behavior, somewhat blocks it.

**Comment by tmtron** [comment]
I also had the issue that one ubunut 20.04 distro would not start.   

I just executed `wsl --shutdown` once and then it worked again, i.e.
* then the (first start) of the distro worked (but it took a while)
* the next starts were then normal (fast, almost instant)
* even after a reboot the distro started fast, as usual
* not sure why this worked: maybe it was just an intermittent resource issue...

**Comment by ANORAK-MATTFLY** [comment]
> wsl --shutdown

I can confirm that this solution worked for me too thanks!!!

**Comment by harem234** [comment]
this worked for me
on WSL 2

wsl -d Debian

**Comment by qhaas** [comment]
Ubuntu 20.04 in WSL2 on Windows 10 Pro for Workstations V20H2
```
> wsl --shutdown
> wsl
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
```

**Comment by Source61** [comment]
Unfortunately none of these worked for me. I got this issue right after using diskpart to compact / reduce file disk since I was running out of disk space and my wsl Debian installation had a 10GB useless file so I deleted it and closed wsl but that didn't help, so I ran diskpart. The disk was successfully compacted after attaching in read-only mode, then I detached and tried to run wsl and got this. I've tried wsl --shutdown, it didn't help. I also tried reattaching drive with diskpart, but then running wsl gave me an error that the disk was already being used by a different application :/

**Comment by Vontei** [comment]
Hi, checking in because I do not see any resolution and the linked issue has been locked. 
Rebooting my system did not work (I'm sure everyone here has tried it). Still stuck and development has halted. So much so that this keeps happening and competitor machines are being considered. 

Are there any known solutions to restarting/reconnecting at this time? 
Do we know the cause?

**Comment by Umoren** [comment]
```bash
wsl --shutdown
wsl
```
worked for me
Ubuntu 20.04

**Comment by ankitgoel739674** [comment]
> After updating to Windows 19555 I get the following error, when starting WSL:
> 
> `A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.`
> 
> ```
> PS C:\Users\add> wsl -l
> Windows Subsystem for Linux Distributions:
> Ubuntu-19.10 (Default)
> docker-desktop
> docker-desktop-data
> PS C:\Users\add> wsl -d Ubuntu-19.10
> A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> PS C:\Users\add> wsl -d docker-desktop
> A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
> PS C:\Users\add>
> ```
> 
> ## Update
> I wanted to try this workaround on a fresh Alpine, [#4371 (comment)](https://github.com/microsoft/WSL/issues/4371#issuecomment-521107556) but neither `/bin` nor `/sbin` were symlinks.

**For me this issue is resolved by starting LxssManager service  :--**
Steps to start "LxssManager" service are  as follows:
1. Press "Windows + R" key to open Run dialog.
2. Type "services.msc" and press "Ok".
3. In services list look for "LxssManager" service and start the service.

**Now, open windows powershell as administrator and type below commands**
  PS C:\WINDOWS\system32> wsl -l -v
  NAME                            STATE           VERSION
**_* Ubuntu-18.04               Running         2_**
  docker-desktop-data    Stopped         2
  docker-desktop             Stopped         2

**If you find state as "Stopped" for your linux distro (ie; ubuntu-18.04, ubuntu 20.04, etc) then type below command**

PS C:\WINDOWS\system32> **wsl -d <distro-name>**

For example:
PS C:\WINDOWS\system32> wsl -d Ubuntu-18.04

This will help to start your linux distro.

Please reach out me if facing any issue on below mail id

techiegeek1994@gmail.com

**Comment by ChrisCodeX** [comment]
After an VSC update in WSL2 Ubuntu 22.04 distro, WSL2 started to fail and showed me this error. I coudn't be able to fix it. So I unistalled and reinstalled Ubuntu 22.04 in WSL2. It worked again, but I lost all my configuration.

**Comment by Iano-theDev** [comment]
wsl --shutdown 
then restart wsl 
that worked for me... on ubntu 20.04

**Comment by VasLem** [comment]
I am able to reproduce this error with WSL2 and Ubuntu 20.04:
A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond.
Error code: Wsl/Service/0x8007274c

**Comment by pradiptaagus** [comment]
After figuring out what is happen I got the solution as follows
1. Open services manager by press `Windows + R` and type `services.msc` and press OK.
2. Find `LxssManager` and stop it.
3. Open the CMD or Powershell and type `wsl -l -v, it` will shows the list of installed distributions. For me, I got Ubuntu.
4. Now run the desired distribution by command `wsl -d Ubuntu`.

**Comment by savemetenminutes** [comment]
Why is this issue closed? This is constantly happening and has been ever since WSL2 came out. It's blatantly obvious that after running WSL2 with memory demanding processeses inside, such as the Docker Desktop WSL2 integration with multiple containers and several open consoles - this issue manifests without fail. Shutting down or restarting WSL2, while being a proprietary "fix" for Microsoft products (especially Windows), is still like saying "it is what it is - deal with it" in a ridiculing and mockful manner. It is quite naive to offer a "solution" that involves shutting down a huge project, which depends on a Docker Compose stack with 20-30 containers, kill the Docker Desktop, shutdown WSL2, obviously quit all processes, which depend on this ecosystem like IDE integrations and consoles, start this whole circus again from scratch, just to hit the same problem 5 minutes later.

**Comment by nicoladj77** [comment]
@savemetenminutes I don't understand if this means that I should give more ram to the WSL2? I have this issue while giving 12GB of ram to the WSL2, how much does it need? Agree with you that this is not a solved issue
